pkg/minikube/translate/translate_test.go

```diff
@@ -26,28 +31,190 @@ func TestSetPreferredLanguage(t *testing.T) {
+			want:  language.French,
+		},
+		{
+			input: "unsuported_lang.utf8",
+			want:  language.French,
+			err:   true,
+		},
 	}
-	for _, tc := range tests {
-		t.Run(tc.input, func(t *testing.T) {
+	for _, test := range tests {
+		t.Run(test.input, func(t *testing.T) {
 			// Set something so that we can assert change.
 			if err := SetPreferredLanguage("is"); err != nil {
 				t.Errorf("unexpected error: %q", err)
 			}
-			if err := SetPreferredLanguage(tc.input); err != nil {
+			if err := SetPreferredLanguage(test.input); err != nil && !test.err {
 				t.Errorf("unexpected error: %q", err)
+			} else if test.err {
+				return
 			}
 
-			want, _ := tc.want.Base()
+			want, _ := test.want.Base()
 			got, _ := GetPreferredLanguage().Base()
 			if got != want {
-				t.Errorf("SetPreferredLanguage(%s) = %q, want %q", tc.input, got, want)
+				t.Errorf("SetPreferredLanguage(%s) = %q, want %q", test.input, got, want)
 			}
 		})
 	}
 
 }
+
+func TestT(t *testing.T) {
+	var tests = []struct {
+		description, input, expected string
+		langDef, langPref            language.Tag
+		translations                 map[string]interface{}
+	}{
+		{
+			description: "empty string not default language",
+			input:       "",
+			expected:    "",
+			langPref:    language.English,
+			langDef:     language.Polish,
+		},
+		{
+			description: "empty string and default language",
+			input:       "",
+			expected:    "",
+			langPref:    language.English,
+			langDef:     language.English,
+		},
+		{
+			description:  "existing translation",
+			input:        "cat",
+			expected:     "kot",
+			langPref:     language.Polish,
+			langDef:      language.English,
+			translations: map[string]interface{}{"cat": "kot"},
+		},
+		{
+			description:  "not existing translation",
+			input:        "cat",
+			expected:     "cat",
+			langPref:     language.Polish,
+			langDef:      language.English,
+			translations: map[string]interface{}{"dog": "pies"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			defaultLanguage = test.langDef
+			preferredLanguage = test.langPref
+			Translations = test.translations
+			got := T(test.input)
+			if test.expected != got {
+				t.Errorf("T(%v) shoud return %v, but got: %v", test.input, test.expected, got)
+			}
+		})
+	}
+}
+
+func TestDetermineLocale(t *testing.T) {
+	var tests = []struct {
+		description string
+		langDef     language.Tag
+		langEnvVar  string
+		expected    string
+	}{
+		{
+			description: "default english",
+			langDef:     language.English,
+			langEnvVar:  "en_US.UTF-8",
+			expected:    "Setting Language to en",
+		},
+		{
+			description: "default polish",
+			langDef:     language.Polish,
+			langEnvVar:  "en_US.UTF-8",
+			expected:    "Setting Language to en",
+		},
+		{
+			description: "preferred polish, no translation",
+			langDef:     language.English,
+			langEnvVar:  "pl_PL.UTF-8",
+			expected:    "Failed to load translation file for pl-PL",
```

Comment:
This test feels like a change-detector test: https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

I'd argue the test could probably be removed. The more important thing is to determine success of a task, rather than what error message is emitted if it fails. This also goes along with the idea of "never inspect the output of error.Error":

https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

Reply:
Hi,
I'm not sure if you are talking about lines `148-153` or whole `129-199`.

Assuming `129-199`
I can remove this if you prefer, but I don't see any better option to test this function.
The function is returning nothing and just printing info for user using `glog`.
In theory, we can invoke the function and ensure it won't panic, but it's not useful in my mind.
I can also refactor this test to just check if `preferredLocal` was configured as expected, but It will cover just a single path.

Assuming `148-153`
this is valid use-case in my mind. By default language is always English(https://github.com/kubernetes/minikube/blob/master/pkg/minikube/translate/translate.go#L32) Some people are using national locales, so If I'm using `LC_ALL=pl_PL` I should get the proper error when a translation cannot be loaded and it's tested here.

Reply:
Hey @n0npax, thanks for adding these tests in. For this one specifically (`129-199`), I agree that we probably shouldn't compare error strings. I think it would be fine to either remove it or to just check if `preferredLocal` is configured as expected.

Reply:
Hi guys,
requested changes implemented. Can you review once again please?